### ISSUE TITLE
[4.2.3] PHP 8.1. Progressive caching. Automatic conversion of false to array is deprecated

### ISSUE THREAD

**Issue by ReLater** [joomla/joomla-cms#38927]
### Steps to reproduce the issue
While debugging, I stumbled across [this known bug](https://github.com/joomla/joomla-cms/issues/38782) and disabled caching for the affected mod_articles_categories, which was followed by a deprecated message that can be reproduced as follows.

- Joomla 4.2.3
- PHP 8.1
- Joomla configuration: Activate progressive caching (see image below).
- !!Clear Joomla Cache!!
- Frontend: Open a page with modules where caching is not disabled (don't know if that matters).
- With hardly configured n3t debug plugin I get

```
Deprecated: Automatic conversion of false to array is deprecated
File: libraries/src/Document/HtmlDocument.php:561
```

See also images.

![grafik](https://user-images.githubusercontent.com/20780646/194778229-7447f6fa-e0e7-4207-bad3-89a0efc6384b.png)

![grafik](https://user-images.githubusercontent.com/20780646/194778233-b9b05f71-f0fd-4bd2-9952-3835c2e174b5.png)

![grafik](https://user-images.githubusercontent.com/20780646/194778254-12f0b88e-f2c0-4d79-85c7-d630cc253c2f.png)

**Comment by richard67** [comment]
`$cbuffer` can only be boolean false if the call to `$cache->get` in line 545 returns false, which is the case when caching is switched off, but then we should not be in that branch of the if statement.

A bit below that line, in line 548 the method returns when `$cbuffer[$hash]` is set.

So it seems that after that line we can assume that the `$cbuffer` was not properly set.

If my assumption is true, it is maybe sufficient and correct just to add a line `$cbuffer = [];` (or `$cbuffer = array();` if keeping the old array syntax like elsewhere in that file) before line 561.

@Fedik What do you think? Can it be solved like this?

**Comment by Fedik** [comment]
Yeap, and I would initialise it on get() https://github.com/joomla/joomla-cms/blob/c93e88c8941eada32c96d8b076b15d4e1fdaf416/libraries/src/Document/HtmlDocument.php#L545
Kind of:
```php
$cbuffer = $cache->get('cbuffer_' . $type) ?: [];
```

**Comment by richard67** [comment]
@Fedik Would be cool if you could make a PR.

**Comment by Fedik** [comment]
Please test the fix #38948
